### PR TITLE
Show Person Already Exists for Person/Candidates

### LINKED ISSUE #1044: Duplicate candidate detection
Create Candidate -> Find Duplicate, allow to update instead
Create Person -> Find Duplicate, allow to update instead
Import -> Find Duplicate, allow to update instead

### PR DESCRIPTION
**Pull request by haiodo**
Show Person Already Exists for Person/Candidates

Closes #1044

Signed-off-by: Andrey Sobolev [haiodo@gmail.com](mailto:haiodo@gmail.com)

### CODE REVIEW

File: plugins/contact/src/index.ts [hunk 1]
```diff
@@ -166,5 +169,70 @@ export default plugin(contactId, {
...
+  })
+
+  const result: FindResult<Person> = []
+
+  for (const c of potentialPersons) {
+    let matches = 0
+    if (c.name === person.name) {
+      matches++
+    }
+    if (c.city === person.city) {
```
Comment (BykhovDenis) [inline plugins/contact/src/index.ts:220]: I'm not sure about city

File: plugins/contact-resources/src/components/CreatePerson.svelte [hunk 2]
```diff
@@ -74,35 +72,81 @@
...
+  $: findPerson(client, { ...object, name: combineName(firstName, lastName) }, channels).then((p) => {
+    matches = p
+  })
 </script>
 
 <Card
   label={contact.string.CreatePerson}
   okAction={createPerson}
-  canSave={firstName.length > 0 && lastName.length > 0}
+  canSave={firstName.length > 0 && lastName.length > 0 && matches.length === 0}
```
Comment (BykhovDenis) [inline plugins/contact-resources/src/components/CreatePerson.svelte:85]: May be we should use status for it and `<StatusControl slot="error" {status} />` like in createApplication?

File: plugins/contact/src/index.ts (hunk 1 above)
Comment (haiodo) [reply plugins/contact/src/index.ts:220]: I think just name is not enough. We could tune this later.

File: plugins/contact-resources/src/components/CreatePerson.svelte (hunk 2 above)
Comment (haiodo) [reply plugins/contact-resources/src/components/CreatePerson.svelte:85]: Nope, it show Person and allow to click.

**Review by BykhovDenis — APPROVED** [review]

**Pull request merged**